Tests/Rewriters/LastCallRewriterTests.cs

```diff
@@ -0,0 +1,206 @@
+Expect.Call (() => _mock1.DoSomething (null, """"))
+  .Constraints (
+    Rhino.Mocks.Constraints.Is.Equal (2) & Rhino.Mocks.Constraints.Is.NotNull(), 
+    Rhino.Mocks.Constraints.Is.Equal (""abc""));")]
+    [TestCase (
+        //language=C#
+        @"
+_mock1.DoSomething (null);
+Console.WriteLine (1);
+LastCall.Constraints (Rhino.Mocks.Constraints.Is.NotNull());",
+        //language=C#
+        @"
+Console.WriteLine (1);
+Expect.Call (() => _mock1.DoSomething (null)).Constraints(Rhino.Mocks.Constraints.Is.NotNull());")]
+    [TestCase (
+        //language=C#
+        @"
+_mock1.DoSomething (null);
+_mock2.DoSomething (null);
+_mock3.DoSomething (null);
+_mock4.DoSomething (null);
+LastCall.Constraints (Rhino.Mocks.Constraints.Is.NotNull());",
+        //language=C#
+        @"
+_mock1.DoSomething (null);
+_mock2.DoSomething (null);
+_mock3.DoSomething (null);
+Expect.Call (() => _mock4.DoSomething (null)).Constraints(Rhino.Mocks.Constraints.Is.NotNull());")]
+    [TestCase (
+        //language=C#
+        @"
+_mock6.DoSomething (null);
+LastCall.IgnoreArguments();",
+        //language=C#
+        @"
+Expect.Call (() => _mock6.DoSomething (null)).IgnoreArguments();")]
+    [TestCase (
+        //language=C#
+        @"
+_mock6.DoSomething (null);
+LastCall.IgnoreArguments();
+_mock3.DoSomething (null);
+Console.WriteLine (1);",
+        //language=C#
+        @"
+Expect.Call (() => _mock6.DoSomething (null)).IgnoreArguments();
+_mock3.DoSomething (null);
+Console.WriteLine (1);")]
+    [TestCase (
+        //language=C#
+        @"
+_mock1.DoSomething (null);
+LastCall.IgnoreArguments();
+_mock2.DoSomething (null);
+LastCall.IgnoreArguments();
+_mock3.DoSomething (null);
+Console.WriteLine (1);
+LastCall.IgnoreArguments();
+Console.WriteLine (1);",
+        //language=C#
+        @"
+Expect.Call (() => _mock1.DoSomething (null)).IgnoreArguments();
+Expect.Call (() => _mock2.DoSomething (null)).IgnoreArguments();
+Console.WriteLine (1);
+Expect.Call (() => _mock3.DoSomething (null)).IgnoreArguments();
+Console.WriteLine (1);")]
+    [TestCase (
+        //language=C#
+        @"
+using(null)
+{
+  _mock6.DoSomething (null);
+  LastCall.IgnoreArguments();
+}",
+        //language=C#
+        @"
+using(null)
+{
+  Expect.Call (() => _mock6.DoSomething (null)).IgnoreArguments();
+}")]
+    [TestCase (
+        //language=C#
+        @"
+using(null)
+{
+  _mock6.DoSomething (null);
+  LastCall.IgnoreArguments();
+}
+
+_mock4.DoSomething (null);
+LastCall.IgnoreArguments();",
+        //language=C#
+        @"
+using(null)
+{
+  Expect.Call (() => _mock6.DoSomething (null)).IgnoreArguments();
+}
+
+_mock4.DoSomething (null);
+LastCall.IgnoreArguments();")]
+    [TestCase (
+        //language=C#
+        @"
+LastCall.Constraints (Rhino.Mocks.Constraints.Is.NotNull());",
+        //language=C#
+        @"
+LastCall.Constraints (Rhino.Mocks.Constraints.Is.NotNull());")]
+    public void Rewrite_LastCall (string source, string expected)
```

Comment:
What about the following:

```csharp
_mock4.DoSomething (null);
_mock6.DoSomething (null);
LastCall.IgnoreArguments();
LastCall.IgnoreArguments();
```

What logic does this result in in Rhino.Mocks, what does the rewriter do?